accounting/src/delta/delta_data_collection/tests/delta_delta_undo_tests.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use DataDelta::{Create, Delete, Modify};
+
+use rstest::rstest;
+
+type ThreeCollections = (
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+    DeltaDataCollection<i32, char>,
+);
+
+#[rstest]
+#[case(make_collections_with_undo(Create('a'), Modify('a', 'b')))]
+#[case(make_collections_with_undo(Modify('a', 'b'), Modify('b', 'c')))]
+#[case(make_collections_with_undo(Modify('a', 'b'), Delete('b')))]
+#[case(make_collections_with_undo(Delete('a'), Create('b')))]
+fn delta_delta_undo_associativity(#[case] collections: ThreeCollections) {
+    {
+        // Delta1 + Delta2 + Undo = Delta1
```

Comment:
When you say Delta1 + Delta2 + Undo, it's not clear which addition is done first. I know this is clear here as it's opposed to the case below, but still, please state it as (Delta1 + Delta2) + Undo